src/utility/program_options.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+#include <string>
+
+/**
+ * @brief The ProgramOptions class is a singleton class with a
+ * fully static interface. It contains all the information the user
+ * provided on the command-line.
+ */
+class ProgramOptions
+{
+public:
+
+    /**
+     * @brief The AlgorithmSelection enum defines the 3 different supported
+     * Algorithms used in this program.
+     *
+     *            BFS: Breadth-First Search
+     * FORD_FULKERSON: The Ford-Fulkerson algorithm implemented using the BFS
+     *    CIRCULATION: The solution for the circulation problem
+     */
+    enum AlgorithmSelection
+    {
+        NONE,
+        BFS,
+        FORD_FULKERSON,
+        CIRCULATION
+    };
+
+    /**
+     * @brief Clears the current ProgramOptions instance
+     *
+     * There isn't much normal use for this method besides
+     * in testing.
+     */
+    static void clear();
+
+    /**
+     * @brief Get the provided filepath of the graph text-file
+     * @return The provided filepath string
+     */
+    static const std::string graph_filepath();
+
+    /**
+     * @brief Get the singleton instance of ProgramOptions
+     *
+     * Check to see if an instance already exists.
+     * If it does return that.
+     * Otherwise, instantiate an instance and return that one.
+     *
+     * @return The current singleton instance of ProgramOptions
+     *
+     */
+    static ProgramOptions* instance();
```

Comment:
There are many ways one can implement a singleton class. I tend to prefer using a static pointer simply because that makes the most sense to me. Plus I don't mind a long-lived object staying in the heap since I want it to stick around until program-close.